### PR TITLE
[topgen] Fix types in test_runs() code

### PR DESCRIPTION
**Pull request by rswarbrick**
Trying to run this file through a type checker, it noticed that the original code misused the `all()` function, to the extent that I don't believe it could ever have worked. The new code does what I think was intended.

@vrozic: I think you wrote the original version of this code. Would you mind taking a quick look to check that this rewrite is what you'd expect?

### CODE REVIEW

**Comment by rswarbrick** [comment]
Oops, that wasn't quite the rewrite that I needed: the types still don't match (because I'm now checking whether a list of booleans is "true" on the last line of `test_runs`). @vrozic: When you have a second, could you let me know what this function is supposed to be doing? I think we've got too many lists in the result somewhere, but I don't want to hack at things without understanding.

**Comment by vrozic** [comment]
Yes, this change looks correct to me.

In short, the function does the following:

- Divides the buffer data into sequences of length SEQ_SIZE_BITS
- For each sequence, it computes 12 values `n_runs[0][i]` and `n_runs[1][i]` for i in range(6).
- Checks that each of the 12 values is within specified thresholds
- Returns `True` only if each of the 12 values if within within thresholds, for every sequence.

File: util/topgen/strong_random.py [hunk 1]
```diff
@@ -134,13 +134,15 @@ def test_runs(self):
...
-            results.append(result)
+            good = True
+            for k in range(6):
+                lo = self.RUNS_LOW_THRESHOLD[k]
+                hi = self.RUNS_HIGH_THRESHOLD[k]
+                if not (lo <= n_runs[0][k] <= hi and
+                        lo <= n_runs[1][k] <= hi):
+                    good = False
+                    break
+            results.append(good)
```
Comment (vrozic) [inline util/topgen/strong_random.py:145]: There are indeed too many lists.
If we move `good = True` outside of the outer loop `for j in range(n_seq)`, we can get rid of the the list `results` and just `return good`

Comment (rswarbrick) [reply util/topgen/strong_random.py]: Good point: that's much cleaner! I've re-pushed with a version that I *think* does what we'd expect.

**Review by vrozic — APPROVED** [review]
This is indeed much cleaner. Thanks for fixing this.
Similar change is possible for functions `test_monobit()`, `test_poker()` and `test_long_run()` but that doesn't need to be in this PR.

**Review by msfschaffner — APPROVED** [review]

**Pull request merged**